src/transformers/trainer.py

```diff
@@ -3784,6 +3784,10 @@ def training_step(
         with self.compute_loss_context_manager():
             loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
 
+        actual_bs = None
+        if "labels" in inputs and isinstance(inputs["labels"], torch.Tensor):
+            actual_bs = inputs["labels"].shape[0]
```

Comment:
`actual_bs` could be defined inside the IF block at line 3819 since it is only used there